opentracing-impl/src/main/java/io/opentracing/AbstractNoopSpan.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+final class AbstractNoopSpan extends AbstractSpan implements NoopSpan, NoopSpanContext {
+
+    static final AbstractNoopSpan INSTANCE = new AbstractNoopSpan("noop");
```

Comment:
perhaps this class should have a different name, like "generic noop span", 'cause it's not really abstract.

Strictly speaking, I think it's not good that these implementation classes are in the core `io.opentracing` package. If they were in a different package, this could've been called simple NoopSpan.